lib/chef/resource/inspec_waiver_file.rb

```diff
@@ -0,0 +1,159 @@
+          action :add
+        end
+      ```
+
+      **Remove an InSpec waiver entry to a given waiver file**:
+
+      ```ruby
+        inspec_waiver_file "my_inspec_control_01" do
+          file '/etc/chef/inspec_waiver.yml'
+          action :remove
+        end
+      ```
+      DOC
+
+      property :control, String,
+        name_property: true,
+        description: "The name of the control being added or removed to the waiver file"
+
+      property :file, String,
+        required: true,
+        description: "The path to the waiver file being modified"
+
+      property :expiration, String,
+        description: "The expiration date of the given waiver - provided in YYYY-MM-DD format",
+        callbacks: {
+          "Expiration date should match the following format: YYYY-MM-DD" => proc { |e|
+            re = Regexp.new('([12]\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\d|3[01]))').freeze
+            re.match?(e)
+          },
+          }
+
+      property :run_test, [true, false],
+        description: "If present and true, the control will run and be reported, but failures in it won’t make the overall run fail. If absent or false, the control will not be run."
+
+      property :justification, String,
+        description: "Can be any text you want and might include a reason for the waiver as well as who signed off on the waiver."
+
+      load_current_value do |desired_state|
+        filename = desired_state.file
+        if ::File.file?(filename) && ::File.readable?(filename) && !::File.zero?(filename)
+          yaml_contents = IO.read(filename)
+          waiver_hash = ::YAML.safe_load(yaml_contents)
+
+          file desired_state.file
+          if waiver_hash.key?("#{desired_state.control}")
+            control desired_state.control
+            if waiver_hash[desired_state.control].key?("expiration_date")
+              expiration waiver_hash[desired_state.control]["expiration_date"]
+            else
+              expiration "1111-11-11"
+            end
+            if waiver_hash[desired_state.control].key?("run")
+              run_test waiver_hash[desired_state.control]["run"]
+            end
+            if waiver_hash[desired_state.control].key?("justification")
+              justification waiver_hash[desired_state.control]["justification"]
+            else
+              justification ""
+            end
+          else
+            control ""
+          end
+        else
+          file ""
+          control ""
+          justification ""
+          expiration "1111-11-11"
+        end
+      end
+
+      action :add do
+        converge_if_changed :file do
+          file "Create Waiver File #{new_resource.file}" do
+            path new_resource.file
+            action :create_if_missing
+          end
+        end
+        converge_if_changed :control, :expiration, :run_test, :justification do
```

Comment:
it is probably better in this case to not try to use `converge_if_changed` but just use the idempotency of the file resource.